lib/plugins/threaded-signing/plugin.c

```diff
@@ -21,41 +21,45 @@
 
 /**
  * This signing plugin sets up a worker thread and calls openssl_sign_hash(), from the worker
- * thread, when there is a new hash to sign. The plugin can only handle one signature at a time,
- * that is, there is no queue of hashes to sign. If the latest signature is not completed by the
- * time of a new request, that new hash is not signed.
+ * thread, when there is a new hash to sign. To handle several signatures at the same time, the
+ * plugin has two buffers. One for incomming hashes and another for outgoing signatures.
  */
 
 #include <assert.h>
 #include <glib.h>
 #include <stdlib.h>  // calloc, malloc, free
 #include <string.h>  // memcpy
+#include <unistd.h>
 
 #include "includes/signed_video_interfaces.h"
 #include "includes/signed_video_openssl.h"
 
-typedef enum {
-  THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN,
-  THREADED_SIGNING_HAS_HASH_TO_SIGN,
-  THREADED_SIGNING_SIGNS_HASH,
-  THREADED_SIGNING_HAS_SIGNATURE,
-  THREADED_SIGNING_ERROR,
-} threaded_signing_plugin_state;
+// This means that the signing plugin can handle a blocked signing hardware up to, for example, 60
+// seconds if the GOP length is 1 second
+#define MAX_BUFFER_LENGTH 60
 
-/* Threaded plugin handle maintaining the thread and locks. Further, stores the hash to sign and the
- * written signature (part of |signature_info|): */
+typedef struct _output_data_t {
+  uint8_t *signature;
+  size_t size;
+  bool signing_error;
+} output_data_t;
```

Comment:
output_data_t is very generic. I'd change it to something like signature_output_data_t (goes for the name starting with underscore as well)